apps/judicial-system/backend/src/app/modules/case/case.service.ts

```diff
@@ -606,48 +606,61 @@ export class CaseService {
     let prosecutorId: string | undefined
     let courtId: string | undefined
 
-    if (caseToCreate.prosecutorNationalId) {
-      const prosecutor = await this.userService.findByNationalId(
-        caseToCreate.prosecutorNationalId,
-      )
-
-      if (!prosecutor || prosecutor.role !== UserRole.PROSECUTOR) {
-        throw new BadRequestException(
-          `User ${prosecutor.id} is not registered as a prosecutor`,
+    try {
+      if (caseToCreate.prosecutorNationalId) {
+        const prosecutor = await this.userService.findByNationalId(
+          caseToCreate.prosecutorNationalId,
         )
-      }
 
-      prosecutorId = prosecutor.id
-      courtId = prosecutor.institution?.defaultCourtId
-    }
+        if (!prosecutor || prosecutor.role !== UserRole.PROSECUTOR) {
+          throw new BadRequestException(
+            `User ${prosecutor.id} is not registered as a prosecutor`,
+          )
+        }
 
-    return this.sequelize
-      .transaction(async (transaction) => {
-        const caseId = await this.createCase(
-          {
-            ...caseToCreate,
-            origin: CaseOrigin.LOKE,
-            creatingProsecutorId: prosecutorId,
-            prosecutorId,
-            courtId,
-          } as InternalCreateCaseDto,
-          transaction,
-        )
+        prosecutorId = prosecutor.id
+        courtId = prosecutor.institution?.defaultCourtId
+      }
 
-        await this.defendantService.create(
-          caseId,
-          {
-            nationalId: caseToCreate.accusedNationalId,
-            name: caseToCreate.accusedName,
-            gender: caseToCreate.accusedGender,
-            address: caseToCreate.accusedAddress,
-          },
-          transaction,
-        )
+      return this.sequelize
+        .transaction(async (transaction) => {
+          const caseId = await this.createCase(
+            {
+              ...caseToCreate,
+              origin: CaseOrigin.LOKE,
+              creatingProsecutorId: prosecutorId,
+              prosecutorId,
+              courtId,
+            } as InternalCreateCaseDto,
+            transaction,
+          )
 
-        return caseId
-      })
-      .then((caseId) => this.findById(caseId))
+          await this.defendantService.create(
+            caseId,
+            {
+              nationalId: caseToCreate.accusedNationalId,
+              name: caseToCreate.accusedName,
+              gender: caseToCreate.accusedGender,
+              address: caseToCreate.accusedAddress,
+            },
+            transaction,
+          )
+
+          return caseId
+        })
+        .then((caseId) => this.findById(caseId))
+    } catch (error) {
```

Comment:
Catching an error here is one way to post error events for LÖKE created cases.

However, this will not catch all possible errors. In particular, failure to call the backend from the xrd-api will not be caught.

A more robust approach would be to catch errors in the xrd-api itself and post events from there. This approach will require either a duplication of the event service in the xrd-api application or moving the event service to a shared library. I recommend using a shared library.

So, consider:
- modifying this PR to handle errors in the xrd-api, or
- creating a refactoring task to move the error handling to the xrd-api later.